Resources/Catalog/FileCatalog.py

```diff
@@ -94,25 +103,33 @@ def w_execute( self, *parms, **kws ):
     successful = {}
     failed = {}
     failedCatalogs = []
-    fileInfo = parms[0]
-    res = checkArgumentFormat( fileInfo )
-    if not res['OK']:
-      return res
-    fileInfo = res['Value']
-    allLfns = fileInfo.keys()
-    parms1 = parms[1:]
-    lfnsFlag = False
+    fileInfo = {}
+    lfnMapDict = {}
+
+    # For methods that pass an LFN argument first do the check and get the
+    # changed LFN mapping
+    lfnsFlag = self.call in FileCatalog.lfn_methods
+    if lfnsFlag:
+      fileInfo = parms[0]
+      res = checkArgumentFormat( fileInfo, generateMap = True )
+      if not res['OK']:
+        return res
+      fileInfo, lfnMapDict = res['Value']
+      kws['NoLFNChecking'] = True
+      allLfns = fileInfo.keys()
+      parms1 = parms[1:]
+
     for catalogName, oCatalog, master in self.writeCatalogs:
 
-      # Skip if metadata related method on pure File Catalog
-      if self.call in FileCatalog.write_meta_methods and not catalogName in self.metaCatalogs:
+      # Skip if the method is not implemented in this catalog
+      if not oCatalog.hasCatalogMethod( self.call ):
```

Comment:
Well, this is not the case right now since the FIleCatalogClient is still inheriting Client which can pass unfound methods to the service. That's why I wanted an explicit list. Also this is a kind of documentation of the client mandate.